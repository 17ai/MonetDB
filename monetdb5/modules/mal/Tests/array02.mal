#extraction of cell positions from arrays

#create array v(i int dimension[0:1:4], payload:int)
v:= array.series(0,1,4,1,1);
vv:= array.series(0,3,12,1,1);
io.print(v,vv);

#create array M(x int dimension[0:2:6], y int dimension[0:1:4], int default 333);
mx:= array.series(0,2,6,4,1);
my:= array.series(0,1,4,1,3);
mv:= array.series(0,1,12,1,1);
io.print(mx,my,mv);

#create a cell identifier vector
#do a point select
x:= bat.new(:oid,:oid);
bat.insert(x, 1@0,0@0);
io.print(x);
y:= bat.new(:oid,:oid);
bat.insert(y, 2@0,0@0);
bat.insert(y, 3@0,1@0);
bat.insert(y, 7@0,2@0);
io.print(y);

#fetch the index column
z1:= array.map(x,v);
io.print(z1);

(zx:bat[:oid,:int],zy:bat[:oid,:int]):= array.map(y,mx,my);
io.print(zx);
io.print(zy);
io.print(zx,zy);
