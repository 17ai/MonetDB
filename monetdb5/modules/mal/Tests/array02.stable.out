stdout of test 'array02` in directory 'modules/mal` itself:


# 21:54:55 >  
# 21:54:55 >   mserver5  --debug=10 --set gdk_nr_threads=0  --set "gdk_dbfarm=/ufs/mk/sciql//Linux/var/MonetDB" --set mapi_open=true --set mapi_port=37601 --set monet_prompt= --trace --forcemito --set mal_listing=2  --dbname=mTests_modules_mal  array02.mal
# 21:54:55 >  

# MonetDB 5 server v11.4.0
# This is an unreleased version
# Serving database 'mTests_modules_mal', using 4 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 7.749 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2011 MonetDB B.V., all rights reserved
# Visit http://monetdb.cwi.nl/ for further information
# Listening for connection requests on mapi:monetdb://eir.ins.cwi.nl:37601/
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
function user.main():void;
#extraction of cell positions from arrays 
#create array v(i int dimension[0:1:4], payload:int) 
    v := array.series(0,1,4,1,1);
    vv := array.series(0,3,12,1,1);
    io.print(v,vv);
#create array M(x int dimension[0:2:6], y int dimension[0:1:4], int default 333); 
    mx := array.series(0,2,6,4,1);
    my := array.series(0,1,4,1,3);
    mv := array.series(0,1,12,1,1);
    io.print(mx,my,mv);
#create a cell identifier vector 
#do a point select 
    x := bat.new(:oid,:oid);
    bat.insert(x,1@0,0@0);
    io.print(x);
    y := bat.new(:oid,:oid);
    bat.insert(y,2@0,0@0);
    bat.insert(y,3@0,1@0);
    bat.insert(y,7@0,2@0);
    io.print(y);
#fetch the index column 
    z1 := array.map(x,v);
    io.print(z1);
    (zx:bat[:oid,:int] ,zy:bat[:oid,:int] ) := array.map(y,mx,my);
    io.print(zx);
    io.print(zy);
    io.print(zx,zy);
end main;
#-------------------------#
# h	t	t	  # name
# void	int	int	  # type
#-------------------------#
[ 0@0,	  0,	  0	  ]
[ 1@0,	  1,	  3	  ]
[ 2@0,	  2,	  6	  ]
[ 3@0,	  3,	  9	  ]
#---------------------------------#
# h	t	t	t	  # name
# void	int	int	int	  # type
#---------------------------------#
[ 0@0,	  0,	  0,	  0	  ]
[ 1@0,	  0,	  1,	  1	  ]
[ 2@0,	  0,	  2,	  2	  ]
[ 3@0,	  0,	  3,	  3	  ]
[ 4@0,	  2,	  0,	  4	  ]
[ 5@0,	  2,	  1,	  5	  ]
[ 6@0,	  2,	  2,	  6	  ]
[ 7@0,	  2,	  3,	  7	  ]
[ 8@0,	  4,	  0,	  8	  ]
[ 9@0,	  4,	  1,	  9	  ]
[ 10@0,	  4,	  2,	  10	  ]
[ 11@0,	  4,	  3,	  11	  ]
#-----------------#
# h	t	  # name
# void	oid	  # type
#-----------------#
[ 1@0,	  0@0	  ]
#-----------------#
# h	t	  # name
# oid	oid	  # type
#-----------------#
[ 2@0,	  0@0	  ]
[ 3@0,	  1@0	  ]
[ 7@0,	  2@0	  ]
#-----------------#
# t	h	  # name
# void	int	  # type
#-----------------#
[ 1@0,	  1	  ]
#-----------------#
# h	t	  # name
# oid	int	  # type
#-----------------#
[ 2@0,	  0	  ]
[ 3@0,	  0	  ]
[ 7@0,	  2	  ]
#-----------------#
# h	t	  # name
# oid	int	  # type
#-----------------#
[ 2@0,	  2	  ]
[ 3@0,	  3	  ]
[ 7@0,	  3	  ]
#-------------------------#
# h	t	t	  # name
# oid	int	int	  # type
#-------------------------#
[ 2@0,	  0,	  2	  ]
[ 3@0,	  0,	  3	  ]
[ 7@0,	  2,	  3	  ]

# 21:54:55 >  
# 21:54:55 >  Done.
# 21:54:55 >  

